pkg/networking/networking.go

```diff
@@ -229,3 +238,11 @@ func (n *networkImpl) Clone() NetworkAccess {
 
 	return clone
 }
+
+func GenerateInteractionUrn() (string, error) {
```

Comment:
Suggestion: 
* Maybe not the best place to add this method, `utils` might be better.
* GenerateInteractionUrn() could take the uuid as a parameter, so that it can be easier re-used.